### PR TITLE
Update GP-Latent example to V4

### PR DESCRIPTION
**Pull request by bwengals**
# Update GP-Latent to V4

Addresses issue #72.  Is a replacement for PR #237.  Upgrades the notebook to V4.  Tried to follow the style guide and passed pre-commit locally, but please let me know what's not correct.  I also cleaned up some of the text and example models a bit.

+ [ ] Notebook follows style guide https://docs.pymc.io/en/latest/contributing/jupyter_style.html
+ [ ] PR description contains a link to the relevant issue: a tracker one for existing notebooks or a proposal one for new notebooks
+ [ ] Check the notebook is not excluded from any pre-commit check: https://github.com/pymc-devs/pymc-examples/blob/main/.pre-commit-config.yaml

### Helpful links
* https://github.com/pymc-devs/pymc-examples/blob/main/CONTRIBUTING.md

### CODE REVIEW

**Review by twiecki — APPROVED** [review]

**Comment by bwengals** [comment]
ty @twiecki

File: myst_nbs/gaussian_processes/GP-Latent.myst.md [hunk 1]
```diff
@@ -6,50 +6,43 @@ jupytext:
...
     jupytext_version: 1.13.7
 kernelspec:
-  display_name: Python (PyMC3 Dev)
+  display_name: pymc
   language: python
-  name: pymc3-dev
+  name: pymc
 ---
 
+```{raw-cell}
```
Comment (OriolAbril) [inline myst_nbs/gaussian_processes/GP-Latent.myst.md:14]: This should not be a raw cell, it should be markdown cell. It is not rendering correctly in the preview due to that: https://pymcio--371.org.readthedocs.build/projects/examples/en/371/gaussian_processes/GP-Latent.html. The title is not shown, neither are the notebook badges and the tags are not recognized.

File: myst_nbs/gaussian_processes/GP-Latent.myst.md [hunk 2]
```diff
@@ -429,11 +424,24 @@ plt.plot(
...
+* Created by [Bill Engels](https://github.com/bwengals) in 2017 ([pymc#1674](https://github.com/pymc-devs/pymc/pull/1674)) 
+* Reexecuted by [Colin Caroll](https://github.com/ColCarroll) in 2019 ([pymc#3397](https://github.com/pymc-devs/pymc/pull/3397))
+* Updated for V4 by Bill Engels in June 2022 ([pymc-examples#237](https://github.com/pymc-devs/pymc-examples/pull/237))
+
 ```{code-cell} ipython3
 %load_ext watermark
 %watermark -n -u -v -iv -w
 ```
+
+```{raw-cell}
```
Comment (OriolAbril) [inline myst_nbs/gaussian_processes/GP-Latent.myst.md:443]: This should not be a raw cell either. There should not be any raw cells in the notebook

File: myst_nbs/gaussian_processes/GP-Latent.myst.md [hunk 3]
```diff
@@ -6,50 +6,43 @@ jupytext:
...
 kernelspec:
-  display_name: Python (PyMC3 Dev)
+  display_name: pymc
   language: python
-  name: pymc3-dev
+  name: pymc
 ---
 
+```{raw-cell}
+(notebook_name)=
```
Comment (OriolAbril) [inline myst_nbs/gaussian_processes/GP-Latent.myst.md:15]: The target should not be literally `notebook_name` but the actual name of the notebook. The text within the parenthesis is what will be used to reference this notebook from other places in the docs, it needs to be unique to this notebook. https://docs.pymc.io/en/latest/contributing/jupyter_style.html#first-cell

**Review by OriolAbril — CHANGES_REQUESTED** [review]
The table of contents is messed up:

image

IMO, the first heading `Gaussian Processes: Latent Variable Implementation` should be the only level 1 heading. Then `Latent variable implementation`, `example 1...`, `example 2...`, `Authors` should be level 2 and so on.

Note: When there is a single level 1 heading at the top of the file, it is excluded from the toc so everything looks better, see for example: https://pymcio--371.org.readthedocs.build/projects/examples/en/371/generalized_linear_models/GLM-simpsons-paradox.html

File: myst_nbs/gaussian_processes/GP-Latent.myst.md (hunk 3 above)
Comment (bwengals) [reply myst_nbs/gaussian_processes/GP-Latent.myst.md]: Hmm right where it says it in that big red box labeled important that I managed to miss

**Comment by bwengals** [comment]
Thanks for the review @OriolAbril!

**Review by OriolAbril — APPROVED** [review]

**Comment by bwengals** [comment]
Hey @OriolAbril, I noticed today that the changes made in this PR were accidentally overwritten by something else a day later (history: https://github.com/pymc-devs/pymc-examples/commits/main/examples/gaussian_processes/GP-Latent.ipynb).  If you look at the version of this notebook in `main` it's old and still has "pymc3" everywhere.  

What's the best way to fix this?  I can open a new PR and copy over the final state of this PR into it, but that sounds like bad git practices.  Any recommendation?

**Comment by bwengals** [comment]
Actually, I found some code that was commented that should be deleted.  I can PR a fix for that that will fix everything else.

**Comment by OriolAbril** [comment]
I think a new PR is perfectly fine, even needed. Having updated notebooks can never be trumped by trying to enforce some git good practices imo. Thanks for catching this by the way.

**Pull request merged**